### PR TITLE
polybar: fixes issue #28387

### PR DESCRIPTION
**Pull request by disassembler**
###### Motivation for this change
This addresses issue #28387 

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by yorickvP** [comment]
How about just bumping this package to git or poking them for a release?

**Comment by vyp** [comment]
It's a one line patch though iirc 🤷

**Comment by yorickvP** [comment]
Maintainer: @afldcr

File: pkgs/applications/misc/polybar/default.nix [hunk 1]
```diff
@@ -39,6 +39,12 @@ stdenv.mkDerivation rec {
       maintainers = [ maintainers.afldcr ];
       platforms = platforms.unix;
     };
```
Comment (afldcr) [inline pkgs/applications/misc/polybar/default.nix]: Can we put a comment above this to mark that this patch should be removed at the next release? It makes it a bit easier on myself as I can't test locally at the moment-- my Linux box is out of commission.

**Comment by afldcr** [comment]
I'm fine with this in the short term provided we put a comment above it explaining why we're patching.

File: pkgs/applications/misc/polybar/default.nix (hunk 1 above)
Comment (disassembler) [inline pkgs/applications/misc/polybar/default.nix]: comment added.

**Comment by afldcr** [comment]
with this comment I say we merge. thanks for your work disassembler.

File: pkgs/applications/misc/polybar/default.nix [hunk 2]
```diff
@@ -39,6 +39,13 @@ stdenv.mkDerivation rec {
...
+    # This patch should be removed with next stable release.
+    patches = [
+      (fetchurl {
```
Comment (cstrahan) [inline pkgs/applications/misc/polybar/default.nix]: You should use `fetchpatch` here, which will cannibalize the patch so the hash doesn't change later (github likes to change the contents subtly, despite the semantics of the patch remaining unchanged).

**Comment by disassembler** [comment]
updated to use fetchpatch.

File: pkgs/applications/misc/polybar/default.nix (hunk 2 above)
Comment (disassembler) [reply pkgs/applications/misc/polybar/default.nix]: done.

**Pull request merged**